CMakeLists.txt

```diff
@@ -115,10 +115,14 @@ elseif(USE_MBEDTLS)
     message(STATUS "Detected clang")
     set(CMAKE_C_FLAGS "-I${CMAKE_CURRENT_SOURCE_DIR}/configs -DMBEDTLS_USER_CONFIG_FILE=\"<config_mbedtls.h>\" ${CMAKE_C_FLAGS}")
     set(CMAKE_CXX_FLAGS "-I${CMAKE_CURRENT_SOURCE_DIR}/configs -DMBEDTLS_USER_CONFIG_FILE=\"<config_mbedtls.h>\" ${CMAKE_CXX_FLAGS}")
-  else()
+  elseif("${CMAKE_C_COMPILER_ID}" MATCHES "GNU")
```

Comment:
We only exclusively test GCC and Clang. So it should be ok to just use `else()`.

Reply:
Added an else clause with the following message:
`message(FATAL_ERROR "Detected unsupported compiler: ${CMAKE_C_COMPILER_ID}")`